hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -241,6 +246,26 @@ private void deleteInstantFile(HoodieInstant instant) {
     return readDataFromPath(detailPath);
   }
 
+  /**
+   * Get the last instant with valid data, and convert this to HoodieCommitMetadata
+   */
+  public Option<Pair<HoodieInstant, HoodieCommitMetadata>> getLastCommitMetadataWithValidData() {
```

Comment:
can we write UT for this method please.

Reply:
especially with MOR table, 
- case1: where latest completed commit is compaction
- case2: has compaction and has more delta commits after compaction
- case3: no compaction and only has delta commits.